src/editor/actions/history-modal-actions.js

```diff
@@ -70,13 +70,28 @@ export function getNotebookRevisionList() {
     dispatch({ type: "GETTING_NOTEBOOK_REVISION_LIST" });
     getRevisionList(getNotebookID(getState()), isLoggedIn(getState()))
       .then(revisionList => {
-        haveLocalAutosave(getState())
-          .then(havePendingChanges => {
+        getLocalAutosave(getState())
+          .then(localAutosaveState => {
+            let hasLocalOnlyChanges = false;
+            if (
+              "revisionContent" in getState().notebookHistory &&
+              "iomd" in localAutosaveState
+            ) {
+              if (
+                revisionList[0].id in getState().notebookHistory.revisionContent
+              ) {
```

Comment:
Looking at this again, we could just rewrite this in a single expression (I know the formatting is wrong):

```js
const hasLocalOnlyChanges = ("iomd" in localAutosaveState  && 
  "revisionContent" in getState().notebookHistory && 
  revisionList.length &&
  revisionList[0].id in getState().notebookHistory.revisionContent &&                   
  localAutosaveState.iomd !== getState().notebookHistory.revisionContent[revisionList[0].id]);
```

This also checks for the edge case where the revision list has a length of 0 (should never happen but who knows).